dhall_syntax/src/core/expr.rs

```diff
@@ -162,8 +177,8 @@ pub enum Builtin {
 }
 
 // Each node carries an annotation. In practice it's either X (no annotation) or a Span.
-#[derive(Debug)]
-pub struct SubExpr<Embed>(Rc<(Expr<Embed>, Option<Span>)>);
+#[derive(Debug, Clone)]
+pub struct SubExpr<Embed>(Box<(Expr<Embed>, Option<Span>)>);
```

Comment:
What's the advantage of `Box` over `Rc`, out of interest?

Reply:
I'm glad you ask ^^
`Rc` made sense back when `SubExpr` was also used for normalization and typechecking, and there was a lot of sharing references to a same subtree. It avoided a lot of cloning. Now we don't do that anymore, so we could use either and the difference would be minimal performance-wise.
The main benefit from using `Box` is that it allows mutation, so we won't need to rebuild the whole `SubExpr` when we resolve imports.
Granted, this is all very much premature optimization, but I have a tendency to do that.

Reply:
Makes sense, thanks for the explanation! :)